src/main/java/exception/DudeException.java

```diff
@@ -0,0 +1,15 @@
+package exception;
+
+/**
+ * Custom exception class for Dude-related exceptions.
+ */
+public class DudeException extends Exception {
```

Comment:
I like how you name your exception, but maybe the naming could be abit more detailed?